go/enclave/enclave.go

```diff
@@ -148,15 +148,25 @@ func NewEnclave(
 		attestationProvider = &DummyAttestationProvider{}
 	}
 
-	// todo (#1053) - this has to be read from the database when the node restarts
-	// first time the node starts we derive the obscuro key from the master seed received after the shared secret exchange
-	logger.Info("Generating the Obscuro key")
-
-	// todo (#1053) - save this to the db
-	enclaveKey, err := gethcrypto.GenerateKey()
+	// attempt to fetch the enclave key from the database
+	enclaveKey, err := storage.GetEnclaveKey()
 	if err != nil {
-		logger.Crit("Failed to generate enclave key.", log.ErrKey, err)
+		if !errors.Is(err, errutil.ErrNotFound) {
+			logger.Crit("Failed to fetch enclave key", log.ErrKey, err)
+		}
+		// enclave key not found - new key should be generated
+		// todo (#1053) - revisit the crypto for this key generation/lifecycle before production
```

Comment:
I left this todo here because it links to the cryptography epic, I think there is work to be done on deriving the key etc but not sure what's involved. Just wanted to get it persisted to unblock the upgrading work.